src/slice/new_iter.rs

```diff
@@ -0,0 +1,495 @@
+//! Okay, this immutable iterator doesn't use any `Access` types, and thus could not be used
+//! on any element with possible mutable access at the same time (or could, but it's a subtle UB).
+//! It could be used without UB only with unaliased memory, and due to this, `.iter()` should produce
+//! a group of `(AliasedIter, UnAliasedIter, AliasedIter)` or use AliasedIter for every `.next()`.
+
+// This should be deleted when all unsafe blocks are used or deleted.
+#![allow(unused_unsafe)]
+
+#[cfg(feature = "std")]
```

Comment:
`core` is always available, because I explicitly link it when `feature = "std"` is present. I don't actually know if the conditional-link in `lib.rs` is necessary anymore, but it doesn't hurt to leave in.

Reply:
Yeah, and most things in std are just re-exports, but what if something really differs in `core/alloc` and `std` but has the same name? I haven't found anything like that yet, but if there will be such a strange inner difference, I export those things from std when that's not [no_std].